pkg/versioning/properties_test.go

```diff
@@ -0,0 +1,122 @@
+package versioning
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"testing"
+
+	"github.com/magiconair/properties"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPropertiesFileGetVersion(t *testing.T) {
+	t.Run("success case", func(t *testing.T) {
+		tmpFolder, err := ioutil.TempDir("", "")
+		if err != nil {
+			t.Fatal("failed to create temp dir")
+		}
+		defer os.RemoveAll(tmpFolder)
+
+		propsFilePath := filepath.Join(tmpFolder, "my.props")
+		ioutil.WriteFile(propsFilePath, []byte("version = 1.2.3"), 0666)
+
+		propsfile := PropertiesFile{
+			path: propsFilePath,
+		}
+		version, err := propsfile.GetVersion()
+
+		assert.NoError(t, err)
+		assert.Equal(t, "1.2.3", version)
+	})
+
+	t.Run("success case - custom version field", func(t *testing.T) {
+		tmpFolder, err := ioutil.TempDir("", "")
+		if err != nil {
+			t.Fatal("failed to create temp dir")
+		}
+		defer os.RemoveAll(tmpFolder)
+
+		propsFilePath := filepath.Join(tmpFolder, "my.props")
+		ioutil.WriteFile(propsFilePath, []byte("customversion = 1.2.3"), 0666)
+
+		propsfile := PropertiesFile{
+			path:         propsFilePath,
+			versionField: "customversion",
+		}
+		version, err := propsfile.GetVersion()
+
+		assert.NoError(t, err)
+		assert.Equal(t, "1.2.3", version)
+	})
+
+	t.Run("error case - file not found", func(t *testing.T) {
+		tmpFolder, err := ioutil.TempDir("", "")
+		if err != nil {
+			t.Fatal("failed to create temp dir")
+		}
+		defer os.RemoveAll(tmpFolder)
+
+		propsFilePath := filepath.Join(tmpFolder, "my.props")
+
+		propsfile := PropertiesFile{
+			path: propsFilePath,
+		}
+		_, err = propsfile.GetVersion()
+
+		assert.Contains(t, fmt.Sprint(err), "failed to load")
+	})
+
+	t.Run("error case - no version found", func(t *testing.T) {
+		tmpFolder, err := ioutil.TempDir("", "")
+		if err != nil {
+			t.Fatal("failed to create temp dir")
+		}
+		defer os.RemoveAll(tmpFolder)
+
+		propsFilePath := filepath.Join(tmpFolder, "my.props")
+		ioutil.WriteFile(propsFilePath, []byte("versionx = 1.2.3"), 0666)
+
+		propsfile := PropertiesFile{
+			path: propsFilePath,
+		}
+		_, err = propsfile.GetVersion()
+
+		assert.EqualError(t, err, "no version found in field version")
+	})
+}
+
+func TestPropertiesFileSetVersion(t *testing.T) {
+	t.Run("success case", func(t *testing.T) {
+		tmpFolder, err := ioutil.TempDir("", "")
+		if err != nil {
+			t.Fatal("failed to create temp dir")
+		}
+		defer os.RemoveAll(tmpFolder)
+
+		propsFilePath := filepath.Join(tmpFolder, "my.props")
+		ioutil.WriteFile(propsFilePath, []byte("version = 0.0.1"), 0666)
+
+		var content []byte
+		propsfile := PropertiesFile{
+			path:      propsFilePath,
+			writeFile: func(filename string, filecontent []byte, mode os.FileMode) error { content = filecontent; return nil },
+		}
+		err = propsfile.SetVersion("1.2.3")
+		assert.NoError(t, err)
+
+		assert.Contains(t, string(content), "version = 1.2.3")
```

Comment:
I wonder if the test code could be a lot smaller, if you used `mock.FilesMock{}`. You have a writeFile function pointer in the struct, but no readFile. Each test currently has to setup a temp-folder... FilesMock{} could be extended to have two methods which return a function with ioutil.ReadFile and ioutil.WriteFile signature, which read/write to the FilesMock instance. Then it would work something like this:
```golang
filesMock := mock.FilesMock{}
filesMock.AddFile("my.props", []byte("version = 0.0.1"))
propsfile := PropertiesFile{
	path:      "my.props",
	readFile:  filesMock.FileReadFunc(),
	writeFile: filesMock.FileWriteFunc(),
}

err = propsfile.SetVersion("1.2.3")

assert.NoError(t, err)
contents, _ := filesMock.FileRead("my.props")
assert.Contains(t, contents, "version = 1.2.3")
```

Reply:
writing would be possible I guess.
Not sure if it is really worth it since for reading we rely on the `properties` lib: `properties.LoadFile()`

Reply:
Ah. That seems to defeat the idea.